server/src/json-api-deserialize.pipe.ts

```diff
@@ -8,7 +9,11 @@ export class JsonApiDeserializePipe implements PipeTransform {
 
     if (type === 'body') {
       return new Deserializer({
-        keyForAttribute: 'snake_case',
+        keyForAttribute: (attribute) => {
+          if (attribute === 'dcp500kpluszone')
+            return 'dcp_500kpluszone';
```

Comment:
Thanks for fixing this! I'm sure this took a lot of head-scratching.

Is there not a more generic way to do this? Using regex or something?

Even a dictionary option would work: you could initialize the serializer/deserializer pipe from the controller with a hash of key-value pairs that behave as a manual translation dictionary for weird, custom one-offs like these.

Reply:
@godfreyyeung  do you think there could end  up being a lot more of these situations? i would add a comment explaining that additional one-offs should also be setup in the serializer. could even mention that if this grows to more, we should probably look into a more generic way.

Reply:
Thanks for the suggestions! Good ideas about parameterizing the serializer/deserializer with custom dictionary of one-offs, and the comments. Both flashed across my mind but I got so wrapped up in sleuthing I forgot!  

So far it seems there is only this one property that starts with a digit after `dcp`. I think if we have more properties, we should definitely take some time to modularize this. I'll definitely add the comments to this PR to start